libgalois/include/katana/EntityTypeManager.h

```diff
@@ -0,0 +1,227 @@
+#pragma once
+
+#include <bitset>
+#include <optional>
+#include <unordered_map>
+#include <vector>
+
+#include "katana/Logging.h"
+
+namespace katana {
+
+/// EntityTypeID uniquely identifies an entity (node or edge) type
+/// EntityTypeID for nodes is distinct from EntityTypeID for edges
+/// This type may either be an atomic type or an intersection of atomic types
+/// EntityTypeID is represented using 8 bits
+using EntityTypeID = uint8_t;
+static constexpr EntityTypeID kUnknownEntityType = EntityTypeID{0};
+static constexpr EntityTypeID kInvalidEntityType =
+    std::numeric_limits<EntityTypeID>::max();
```

Comment:
Please document these. Having two sentinels means it's extra import to make if very clear what they mean.